pkg/graph/resolvers/trb_request_feedback.go

```diff
@@ -0,0 +1,87 @@
+package resolvers
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+
+	"github.com/cmsgov/easi-app/pkg/appcontext"
+	"github.com/cmsgov/easi-app/pkg/models"
+	"github.com/cmsgov/easi-app/pkg/storage"
+)
+
+// CreateTRBRequestFeedback creates a TRBRequestFeedback in the database
+func CreateTRBRequestFeedback(ctx context.Context, store *storage.Store, feedback *models.TRBRequestFeedback) (*models.TRBRequestFeedback, error) {
+	feedback.CreatedBy = appcontext.Principal(ctx).ID()
+
+	form, err := store.GetTRBRequestFormByTRBRequestID(ctx, feedback.TRBRequestID)
+	if err != nil {
+		return nil, err
+	}
+
+	trb, err := store.GetTRBRequestByID(appcontext.ZLogger(ctx), feedback.TRBRequestID)
+	if err != nil {
+		return nil, err
+	}
+
+	// If status is edits requested
+	if feedback.Action == models.TRBFeedbackActionRequestEdits {
+		// Update the TRB form status to in progress
+		formChanges := map[string]interface{}{
+			"status": models.TRBFormStatusInProgress,
+		}
+		err = ApplyChangesAndMetaData(formChanges, form, appcontext.Principal(ctx))
+		if err != nil {
+			return nil, err
+		}
+
+		_, err = store.UpdateTRBRequestForm(ctx, form)
+		if err != nil {
+			return nil, err
+		}
+
+		// Update TRB feedback status to "edits requested"
+		trbChanges := map[string]interface{}{
+			"feedbackStatus": models.TRBFeedbackStatusEditsRequested,
+		}
+		err = ApplyChangesAndMetaData(trbChanges, trb, appcontext.Principal(ctx))
+		if err != nil {
+			return nil, err
+		}
+
+		_, err = store.UpdateTRBRequest(appcontext.ZLogger(ctx), trb)
+		if err != nil {
+			return nil, err
+		}
+	} else if feedback.Action == models.TRBFeedbackActionReadyForConsult {
+		// Update TRB feedback status to "edits requested"
+		trbChanges := map[string]interface{}{
+			"feedbackStatus": models.TRBFeedbackStatusCompleted,
+		}
+		err = ApplyChangesAndMetaData(trbChanges, trb, appcontext.Principal(ctx))
+		if err != nil {
+			return nil, err
+		}
+
+		_, err = store.UpdateTRBRequest(appcontext.ZLogger(ctx), trb)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
Where did we land on calculated statuses (at time of retrieval) vs. stored statuses (at time of action)?

I thought we'd landed on storing stuff like `feedbackRequested: true` and letting the status get calculated off that, rather than storing a discrete `status` column?

I mainly ask this because I want to make sure we're considering the multiple different statuses that task list sections might have, and trying to represent all of that in a `status` column on the request itself. (i.e. - we might want to return a status of "[Scheduled](https://www.figma.com/file/iGatgTnlzTuqW9CQFYHDam/TRB-Intake-e2e?node-id=18%3A7378&t=bbs6Gv4V6Udti6iu-0)" or "[Completed](https://www.figma.com/file/iGatgTnlzTuqW9CQFYHDam/TRB-Intake-e2e?node-id=18%3A8339&t=bbs6Gv4V6Udti6iu-0)" for the "4. Attend the TRB consult session" step based on the time it was scheduled, so this is something we'd probably want to calculate)

Reply:
@ClayBenson94 Is this documented anywhere? I remember we talked about this in a meeting a while ago, but don't recall what the specific conclusions were. Just let me know how you want it to be.